plugin.go

```diff
@@ -158,19 +162,18 @@ func NewCompiledPlugin(
 	//  - If there is only one module in the manifest then that is the main module by default
 	//  - Otherwise the last module listed is the main module
 
-	modules := map[string]wazero.CompiledModule{}
+	foundMain := false
 	for i, wasm := range manifest.Wasm {
 		data, err := wasm.ToWasmData(ctx)
 		if err != nil {
 			return nil, err
 		}
 
-		_, mainExists := modules["main"]
-		if data.Name == "" || i == len(manifest.Wasm)-1 && !mainExists {
+		if data.Name == "" || i == len(manifest.Wasm)-1 && !foundMain {
 			data.Name = "main"
 		}
 
-		_, okm := modules[data.Name]
+		_, okm := p.modules[data.Name]
 
 		if data.Name == "extism:host/env" || okm {
```

Comment:
There can be a case, when there are 2 modules with empty name, so both will become `main` which is an error
```suggestion
		if data.Name == "extism:host/env" || okm || (foundMain && data.Name == "main") {
```

Also, I think there should be a check that each module has non-empty name